internal/services/compute/migration/managed_disk_v0_to_v1.go

```diff
@@ -0,0 +1,357 @@
+package migration
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2022-03-02/disks"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ pluginsdk.StateUpgrade = ManagedDiskV0ToV1{}
+
+type ManagedDiskV0ToV1 struct{}
+
+func (ManagedDiskV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		oldId, err := disks.ParseDiskIDInsensitively(rawState["id"].(string))
+		if err != nil {
+			return rawState, err
+		}
+
+		rawState["id"] = oldId.ID()
+		return rawState, nil
+	}
+}
+
+func (ManagedDiskV0ToV1) Schema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"location": azure.SchemaLocation(),
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"storage_account_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(disks.DiskStorageAccountTypesStandardLRS),
+				string(disks.DiskStorageAccountTypesStandardSSDZRS),
+				string(disks.DiskStorageAccountTypesPremiumLRS),
+				string(disks.DiskStorageAccountTypesPremiumVTwoLRS),
+				string(disks.DiskStorageAccountTypesPremiumZRS),
+				string(disks.DiskStorageAccountTypesStandardSSDLRS),
+				string(disks.DiskStorageAccountTypesUltraSSDLRS),
+			}, false),
+			DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
State migrations only require the Type and Required/Optional/Computed for most fields, things like List of Strings also require the Elem Type, so these fields like these can be removed from the `Schema()`

Reply:
I see, thanks for the information! I've removed the meta-arguments other than Required/Optional/Computed.